controllers/update_test.go

```diff
@@ -480,8 +480,12 @@ func TestImageAutomationReconciler_e2e(t *testing.T) {
 
 		t.Run("PushSpec", func(t *testing.T) {
 			// Clone the repo locally.
-			localRepo, err := clone(cloneLocalRepoURL, "origin", branch)
+			localRepo, err := clone(cloneLocalRepoURL, branch)
 			g.Expect(err).ToNot(HaveOccurred(), "failed to clone git repo")
+			defer localRepo.Free()
+			origin, err := localRepo.Remotes.Lookup("origin")
+			g.Expect(err).ToNot(HaveOccurred(), "failed to clone git repo")
+			defer origin.Free()
```

Comment:
If this origin lookup needs to be performed after a clone to make sure clone happened successfully, maybe it'd be better to move this check within the clone function.